ocaml/xapi/rpm.ml

```diff
@@ -157,9 +159,41 @@ module Pkg = struct
     | None, None ->
         EQ
 
+  let compare_version_segment s1 s2 =
+    match (s1, s2) with
+    | Int i1, Int i2 ->
+        Int.compare i1 i2 |> order_of_int
+    | Str s1, Str s2 ->
+        String.compare s1 s2 |> order_of_int
+    | Tilde, Tilde ->
+        EQ
+    | Int _, Str _ ->
+        GT
+    | Str _, Int _ ->
+        LT
+    | Tilde, _ ->
+        LT
+    | _, Tilde ->
+        GT
+
+  let split_version_string =
+    let r = Re.Posix.compile_pat {|[a-zA-Z]+|[0-9]+|~|} in
```

Comment:
Do we have to detect when a version string contains illegal characters? For example, the above regex does not allow a underscore to be part of the string.

Reply:
All illegal characters expect for `[alphabetic]+, [digit]+ and ~` are acted as delimiter:
```
1.2#3_45abc -> ["1"; "2"; "3"; "45"; "abc"]
1.2.3α45 -> ["1"; "2"; "3"; "45"]
12汉字45 -> ["12"; "45"]
```